### ISSUE TITLE
Waiting for an auto-merge...

### ISSUE THREAD

**Issue by abernix** [renovatebot/config-help#58]
👋 

I'm currently trying to determine why the auto-merge is not occurring on https://github.com/apollographql/apollo/pull/171.

Ultimately, I think it should be obeying [these rules](https://github.com/meteor/renovate-config-mdg-docs/blob/master/package.json) (via [this](https://github.com/apollographql/renovate-config-apollo-docs/blob/master/package.json)).

Help very much appreciated in diagnosing!

**Comment by rarkins** [comment]
Looking in the logs, I can see:
```
{"level":20,"repository":"apollographql/apollo","branch":"renovate/meteor-theme-hexo-1.x","dependencies":["meteor-theme-hexo"],"automerge":false,"automergeType":"pr","automergeComment":"automergeComment","msg":"Checking #171 for automerge","time":"2018-06-20T11:19:12.604Z"}
{"level":20,"repository":"apollographql/apollo","branch":"renovate/meteor-theme-hexo-1.x","dependencies":["meteor-theme-hexo"],"msg":"No automerge","time":"2018-06-20T11:19:12.604Z"}
```

(FYI you can now see the same logs if you log in at https://renovatebot.com/dashboard)

So Renovate thinks this branch has no automerge.

The repo's configuration has:
```
{
  "pathRules": [
    {
      "paths": ["docs/package.json"],
      "extends": ["apollo-docs"]
    }
  ]
}
```

So next step is to make sure this rule is being applied. The logs show a migration from `pathRules` to `packgeRules` but this shouldn't effect anything.

Here is the particular packageRule that I think @abernix is referring to: https://github.com/meteor/renovate-config-mdg-docs/blob/master/package.json#L28

Looking at the branch name, it seems that the rule is not being applied. I'll dig into that next.

**Comment by rarkins** [comment]
@abernix it looks to me like it's a packageRules "inception" problem, e.g. packagerules inside packageRules. So needs a fix/enhancement to work - I'll see if it's a quick one.

**Comment by rarkins** [comment]
> Could it be related to a need for having `"packageFiles": ["docs/package.json"],` since there's no `package.json` in the root of the repository?

Yes. Although it's a valid use case/configuration, I am not aware of anyone else having a need to control the `extends` by path, so this is rare or perhaps unique.

> I can't help but feel like something might have changed with Renovate though because it seems [this PR](https://github.com/apollographql/apollo-server/pull/1087) (on another repository) with a [similar `renovate.json` configuration](https://github.com/apollographql/apollo-server/blob/f6fd3b22311a305748a04953cbed2430024ee0dc/renovate.json#L10-L15).  I tried checking on the Dashboard, but I think it's too far back in history.

It's quite possible that a refactor ~1 month ago may have broken this, although I'd be surprised/impressed if it worked completely before without being *deliberately* supported.

FYI I think I have a fix for this, will do some testing then push it live. If it works, it will result in the existing PR being autoclosed because the "proper" PR should be using a different branch name (`renovate/hexo-related`). Do you mind that minor "noise" if it means we can verify it's fixed?

**Comment by abernix** [comment]
I don't mind the noise — thanks for looking at this  The new branch seems to be open now, but I haven't seen a PR surface yet.

I certainly understand that controlling the `extends` via path is likely rare.  We used to have it configured differently on this repo as well, but changed that as a result of our conversations in https://github.com/renovatebot/config-help/issues/23#issuecomment-377202632.

I'm really looking forward to using the dashboard to help diagnose things in the future - I just used it now to determine that the new branch (created as a result of your change above) hadn't opened the PR but would within the hour (presumably) because of `prCreation: 'not-pending'` and `prNotPendingHours` not being met yet (too early)!

Too easy!  (Again, thanks!)
